tests/main/fake-netplan-apply/task.yaml

```diff
@@ -6,29 +6,38 @@ details: |
 environment:
     NETPLAN: io.netplan.Netplan
 
-# don't run on non-ubuntu systems where we won't have netplan
-# and also don't run on ubuntu-core where we can't create the fake netplan D-Bus
-# service
+# only run on ubuntu lts's which match ubuntu core series, we don't run this on
+# core since there we can't setup the fake dbus service, but we have a core
+# specific spread test in tests/core/netplan which uses the real netplan
 systems:
-    - -debian-*
-    - -fedora-*
-    - -ubuntu-14*
-    - -ubuntu-core-*
-    - -opensuse-*
-    - -arch-*
-    - -amazon-*
-    - -centos-*
-    # TODO: re-enable this when network-setup-control is updated to allow
-    # running netplan directly from the base snap and we can update this test to
-    # just using a simple shell script snap instead of building netplan from
-    # source, as currently that is broken on xenial with
-    # test-snapd-netplan-apply for some reason, see https://github.com/anonymouse64/test-snapd-netplan-apply/issues/1
-    - -ubuntu-16*
+    - ubuntu-16.04*
+    - ubuntu-18.04*
+    - ubuntu-20.04*
 
 prepare: |
+    # build the netplan snap for this system
     #shellcheck source=tests/lib/snaps.sh
     . "$TESTSLIB"/snaps.sh
-    snap install test-snapd-netplan-apply --edge
+
+    if os.query is-xenial; then
```

Comment:
shouldn't we test all the combinations here as well? or we don't because we do that on core?  anyway testing on classic with matching base seems a bit odd as it's not a typical scenario, if we think the matrix is tested elsewhere we might test here just with the base: core18 one

Reply:
The core test appears to try all bases on each ubuntu core release, so I agree that it may be useful to just run this one with a single base to save both the execution time and leave less room for random failures.

Reply:
Yes, I would rather keep the matrix of testing all combos just on UC, since that's where netplan is most likely to be used, and also specifically where our customers are going to use netplan, on Classic ubuntu there are many other options to configure networking, but netplan is supposed to be the first class citizen on core. 

I will update the classic test to use core18 though

Reply:
actually after looking at this again, do we have an issue with the current way it is written to test core on 16.04, core18 on 18.04 and core20 on 20.04? Seems reasonable to me and we don't lose much time having the switch to use a different base snap for each system, so if there's no opposition to testing the version of netplan matching the host OS I would rather keep the classic test as-is

Reply:
it's longer and gives the impression that things have to match for a reason (maybe is just a matter of commenting regarding the latter)

Reply:
I updated the comment above the `systems` line, please take a look and see if it makes sense, if you still want to just test a single version of the netplan client library, I can do that too of course if really desired